src/schema/feeds.ts

```diff
@@ -686,6 +688,66 @@ const applyFeedPaging = (
   return newBuilder;
 };
 
+const feedPageGeneratorWithPin: PageGenerator<
+  GQLPost,
+  FeedArgs,
+  FeedPage,
+  unknown
+> = {
+  connArgsToPage: ({ first, after }: FeedArgs) => {
+    const cursor = getCursorFromAfter(after);
+    const limit = Math.min(first || 30, 50) + 1;
+
+    return cursor
+      ? { limit, timestamp: new Date(parseInt(cursor)) }
+      : { limit };
+  },
+  nodeToCursor: (_, __, node) => {
+    const createdAt = new Date(node.createdAt);
+    const timestamp =
+      node.type === PostType.Welcome
+        ? changeYearToNextYear(createdAt)
+        : createdAt;
+
+    return base64(`time:${new Date(timestamp).getTime()}`);
+  },
+  hasNextPage: (page, nodesSize) => page.limit === nodesSize,
+  hasPreviousPage: (page) => !!page.timestamp,
+  transformNodes: (page, nodes) => nodes.slice(0, page.limit - 1),
+};
+
+const applyFeedPagingWithPin = (
+  ctx: Context,
+  page: FeedPage,
+  builder: SelectQueryBuilder<Post>,
+  alias,
+): SelectQueryBuilder<Post> => {
+  const priority = `CASE WHEN ${alias}.type = '${PostType.Welcome}' THEN 1 ELSE 0 END`;
+  let newBuilder = builder
+    .addSelect(`${alias}."createdAt" as "createdAt"`)
```

Comment:
Let's add `createdAt` as required field in GraphORM instead. I want to avoid selecting the same field twice in this case.

Reply:
Just checked the graphorm, looks to have the `createdAt` already, I'll try to remove this.